pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -2050,3 +2051,36 @@ func TestWorkloadInstanceHandlerMultipleEndpoints(t *testing.T) {
 		}
 	}
 }
+
+func TestKubeEndpointsControllerOnEvent(t *testing.T) {
+	testCases := []struct {
+		mode      EndpointMode
+		tombstone cache.DeletedFinalStateUnknown
+	}{
+		{
+			mode: EndpointsOnly,
+			tombstone: cache.DeletedFinalStateUnknown{
+				Key: "namespace/name",
+				Obj: &coreV1.Endpoints{},
+			},
+		},
+		{
+			mode: EndpointSliceOnly,
+			tombstone: cache.DeletedFinalStateUnknown{
+				Key: "namespace/name",
+				Obj: &discovery.EndpointSlice{},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
You can create single test case and iterate the mode like this https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/controller_test.go#L88

Reply:
Right, but the tombstone object also has a different type depending on the mode. So this seemed easier.